### PR TITLE
chore!: bump sdk and core

### PR DESCRIPTION
**Pull request by evan-forbes**
## Overview

bumps to our forks of the cosmos-sdk and comet to their latest versions. This is consensus breaking because an event was modified in the sdk, although that event may or may not have gotten hit on mocha-2 already.

blocked by https://github.com/celestiaorg/cosmos-sdk/pull/326, https://github.com/celestiaorg/cosmos-sdk/pull/324, https://github.com/celestiaorg/celestia-core/pull/1021

### CODE REVIEW

**Review by cmwaters — DISMISSED** [review]

File: app/prepare_proposal.go [hunk 1]
```diff
@@ -14,11 +14,9 @@ import (
 // generates the data root for the proposal block and passes it back to
 // tendermint via the BlockData.
 func (app *App) PrepareProposal(req abci.RequestPrepareProposal) abci.ResponsePrepareProposal {
-	sdkCtx, err := app.NewProcessProposalQueryContext()
-	if err != nil {
-		panic(err)
-	}
-
+	// create a context using a branch of the state and loaded using the
```
Comment (SweeXordious) [inline app/prepare_proposal.go:17]: [To understand more][non-blocking]
What do you mean by `a branch of the state`?

Comment (evan-forbes) [reply app/prepare_proposal.go:17]: just that under the hood we're calling this function https://github.com/celestiaorg/cosmos-sdk/blob/24903f4a983911f708fbf138a895ea4abc5dc752/store/rootmulti/store.go#L462-L476, which gives us a fork of the state that we operate on without affecting any other branches

File: app/square_size.go [hunk 2]
```diff
@@ -14,7 +14,7 @@ func (app *App) GovSquareSizeUpperBound(ctx sdk.Context) int {
 	// of the sdk and comet that have full support of PreparePropsoal, although
 	// celestia-app does not currently use those. see this PR for more
 	// details https://github.com/cosmos/cosmos-sdk/pull/14505
```
Comment (rootulp) [inline app/square_size.go:16]: is this comment still accurate? Specifically the "are not `>= 1`." part?

**Review by rootulp — APPROVED** [review]

File: app/square_size.go [hunk 3]
```diff
@@ -9,12 +9,12 @@ import (
...
-	// PrepareProposal when heights are not >= 1. This is remedied in versions
-	// of the sdk and comet that have full support of PreparePropsoal, although
-	// celestia-app does not currently use those. see this PR for more
-	// details https://github.com/cosmos/cosmos-sdk/pull/14505
-	if ctx.BlockHeader().Height == 0 {
+	// 64. This is due to our fork of the sdk not initializing state before
+	// BeginBlock of the first block. This is remedied in versions of the sdk
+	// and comet that have full support of PreparePropsoal, although
+	// celestia-app does not currently use those. see this PR for more details
+	// https://github.com/cosmos/cosmos-sdk/pull/14505
```
Comment (cmwaters) [inline app/square_size.go:16]: Are we going to port this across?

**Review by cmwaters — APPROVED** [review]

**Review by rootulp — APPROVED** [review]

**Pull request merged**